### ISSUE TITLE
1.12-RC1: 63 deprecation warnings while building with -Wdeprecated-declarations

### ISSUE THREAD

**Issue by paulgevers** [lwindolf/liferea#454]
Hi,

In Debian, the default build flags include -Wdeprecated-declarations, which shows 63 warnings in the build log¹. This bug is just to warn you about this, in case you don't build with that option yourselves.

¹ https://buildd.debian.org/status/fetch.php?pkg=liferea&arch=all&ver=1.12~rc1-6&stamp=1476640882

Paul

```
paul@testavoira ~ $ grep "is deprecated" /tmp/log
feed_list_view.c:441:2: warning: 'gtk_toggle_action_get_active' is deprecated [-Wdeprecated-declarations]
itemview.c:457:2: warning: 'gtk_widget_reparent' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:353:2: warning: 'gtk_action_set_sensitive' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:353:2: warning: 'gtk_action_group_get_action' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:359:2: warning: 'gtk_action_group_set_sensitive' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:360:2: warning: 'gtk_action_group_set_sensitive' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:361:2: warning: 'gtk_action_group_set_sensitive' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:367:2: warning: 'gtk_action_group_set_sensitive' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:373:2: warning: 'gtk_action_set_sensitive' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:373:2: warning: 'gtk_action_group_get_action' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:374:2: warning: 'gtk_action_set_sensitive' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:374:2: warning: 'gtk_action_group_get_action' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:380:2: warning: 'gtk_action_set_sensitive' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:380:2: warning: 'gtk_action_group_get_action' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:381:2: warning: 'gtk_action_set_sensitive' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:381:2: warning: 'gtk_action_group_get_action' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1216:2: warning: 'gtk_ui_manager_new' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1218:2: warning: 'gtk_action_group_new' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1219:2: warning: 'gtk_action_group_set_translation_domain' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1220:2: warning: 'gtk_action_group_add_actions' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1221:2: warning: 'gtk_action_group_add_toggle_actions' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1222:2: warning: 'gtk_action_group_add_radio_actions' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1223:2: warning: 'gtk_action_group_add_toggle_actions' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1224:2: warning: 'gtk_ui_manager_insert_action_group' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1226:2: warning: 'gtk_action_group_new' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1227:2: warning: 'gtk_action_group_set_translation_domain' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1228:2: warning: 'gtk_action_group_add_actions' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1229:2: warning: 'gtk_ui_manager_insert_action_group' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1231:2: warning: 'gtk_action_group_new' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1232:2: warning: 'gtk_action_group_set_translation_domain' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1233:2: warning: 'gtk_action_group_add_actions' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1234:2: warning: 'gtk_ui_manager_insert_action_group' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1236:2: warning: 'gtk_action_group_new' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1237:2: warning: 'gtk_action_group_set_translation_domain' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1238:2: warning: 'gtk_action_group_add_actions' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1239:2: warning: 'gtk_ui_manager_insert_action_group' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1241:2: warning: 'gtk_action_group_new' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1242:2: warning: 'gtk_action_group_set_translation_domain' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1243:2: warning: 'gtk_action_group_add_actions' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1244:2: warning: 'gtk_ui_manager_insert_action_group' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1246:2: warning: 'gtk_ui_manager_get_accel_group' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1252:2: warning: 'gtk_ui_manager_add_ui_from_string' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1256:3: warning: 'gtk_ui_manager_add_ui_from_string' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1259:3: warning: 'gtk_ui_manager_add_ui_from_string' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1263:2: warning: 'gtk_ui_manager_get_widget' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1264:2: warning: 'gtk_ui_manager_get_widget' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1270:2: warning: 'gtk_ui_manager_get_widget' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1271:2: warning: 'gtk_ui_manager_get_widget' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1272:2: warning: 'gtk_ui_manager_get_widget' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1273:2: warning: 'gtk_ui_manager_get_widget' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1274:2: warning: 'gtk_ui_manager_get_widget' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1453:2: warning: 'gtk_action_group_get_action' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1453:2: warning: 'gtk_radio_action_get_type' is deprecated [-Wdeprecated-declarations]
liferea_shell.c:1454:2: warning: 'gtk_radio_action_set_current_value' is deprecated [-Wdeprecated-declarations]
preferences_dialog.c:691:2: warning: 'gtk_alignment_new' is deprecated [-Wdeprecated-declarations]
preferences_dialog.c:692:2: warning: 'gtk_alignment_set_padding' is deprecated [-Wdeprecated-declarations]
preferences_dialog.c:692:2: warning: 'gtk_alignment_get_type' is deprecated [-Wdeprecated-declarations]
itemlist.c:635:2: warning: 'gtk_radio_action_get_current_value' is deprecated [-Wdeprecated-declarations]
net.c:286:2: warning: 'soup_session_async_new_with_options' is deprecated: Use 'soup_session_new' instead [-Wdeprecated-declarations]
net.c:292:2: warning: 'soup_session_async_new_with_options' is deprecated: Use 'soup_session_new' instead [-Wdeprecated-declarations]
render.c:205:2: warning: 'gtk_widget_get_style' is deprecated: Use 'gtk_widget_get_style_context' instead [-Wdeprecated-declarations]
render.c:231:3: warning: 'gdk_color_free' is deprecated: Use 'gdk_rgba_free' instead [-Wdeprecated-declarations]
render.c:239:3: warning: 'gdk_color_free' is deprecated: Use 'gdk_rgba_free' instead [-Wdeprecated-declarations]
```

**Comment by lwindolf** [comment]
I'm aware of the warnings (using similar compile flags). Still the deprecation speed of GTK+ is so high, porting never keeps pace. Usually fixes only when deprecation actually happens in recent Ubuntu/Fedora versions.

**Comment by lwindolf** [comment]
Just an update: @leiaz is currently working on the GtkUiManager warnings.

**Comment by Leiaz** [comment]
Here is what is left : 
```
../../../src/ui/liferea_shell.c:246:3: warning: ‘gdk_screen_width’ is deprecated [-Wdeprecated-declarations]
../../../src/ui/liferea_shell.c:247:4: warning: ‘gdk_screen_width’ is deprecated [-Wdeprecated-declarations]
../../../src/ui/liferea_shell.c:251:3: warning: ‘gdk_screen_height’ is deprecated [-Wdeprecated-declarations]
../../../src/ui/liferea_shell.c:252:4: warning: ‘gdk_screen_height’ is deprecated [-Wdeprecated-declarations]
../../../src/ui/liferea_shell.c:312:6: warning: ‘gdk_screen_width’ is deprecated [-Wdeprecated-declarations]
../../../src/ui/liferea_shell.c:313:6: warning: ‘gdk_screen_height’ is deprecated [-Wdeprecated-declarations]
../../../src/webkit/webkit.c:163:2: warning: ‘gdk_screen_get_width’ is deprecated [-Wdeprecated-declarations]
../../../src/webkit/webkit.c:163:2: warning: ‘gdk_screen_get_height’ is deprecated [-Wdeprecated-declarations]
../../../src/webkit/webkit.c:164:2: warning: ‘gdk_screen_get_width_mm’ is deprecated [-Wdeprecated-declarations]
../../../src/webkit/webkit.c:164:2: warning: ‘gdk_screen_get_height_mm’ is deprecated [-Wdeprecated-declarations]
```
The replacements for those are [GdkMonitor](https://developer.gnome.org/gdk3/stable/GdkMonitor.html) functions that are only available since Gdk 3.22. It will have to wait if we want to keep working on Debian 8.

```
../../../src/ui/popup_menu.c:78:2: warning: ‘gtk_menu_popup’ is deprecated: Use '(gtk_menu_popup_at_widget, gtk_menu_popup_at_pointer, gtk_menu_popup_at_rect)' instead [-Wdeprecated-declarations]
../../../src/webkit/liferea_web_view.c:221:2: warning: ‘gtk_menu_popup’ is deprecated: Use '(gtk_menu_popup_at_widget, gtk_menu_popup_at_pointer, gtk_menu_popup_at_rect)' instead [-Wdeprecated-declarations]
../../src/browser.c:125:3: warning: ‘gtk_show_uri’ is deprecated: Use 'gtk_show_uri_on_window' instead [-Wdeprecated-declarations]
```
The replacements requires Gtk 3.22.

```
../../src/render.c:205:2: warning: ‘gtk_widget_get_style’ is deprecated: Use 'gtk_widget_get_style_context' instead [-Wdeprecated-declarations]
../../src/render.c:231:3: warning: ‘gdk_color_free’ is deprecated: Use 'gdk_rgba_free' instead [-Wdeprecated-declarations]
../../src/render.c:239:3: warning: ‘gdk_color_free’ is deprecated: Use 'gdk_rgba_free' instead [-Wdeprecated-declarations]
```
This one will require more changes as the style context doesn't work exactly like the old style.

**Comment by lwindolf** [comment]
A lot of stuff has being fixed and there is a separate task for fixing the gdk_screen deprecations.

Closing this to avoid ticket rot.
